server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -615,6 +616,9 @@ private void parseValueAddFields(ParseContext context, String value, String fiel
             }
             if (fieldName.equals(fieldType().name())) {
                 context.doc().add(field);
+                if (fieldType().hasDocValues()) {
```

Comment:
Yes, we can! And let's found out why test fails :)

Reply:
> If you move it to the previous place, JUnit tests will be failed. Can we move it?

@marevol because now with the if statement of getField will become true, it will add SortedSetDocValuesField into the doc(), and the indexableField length will be get changed in here in https://github.com/opensearch-project/OpenSearch/blob/28d5948d4fb5acd55fc686c05058c85b12b95c18/server/src/test/java/org/opensearch/index/mapper/FlatObjectFieldMapperTests.java#LL110C37-L110C37 You can try to print out the length and also the indexable fields and see if that's expected.

Reply:
Although I might not understand it correctly, if moving the code to the previous location like [this](https://github.com/marevol/OpenSearch/commit/415b4bbcea20afd994cf66a6d1b5b7fe77cc46bc), the length will be 1.
However, [the moved code](https://github.com/marevol/OpenSearch/blob/415b4bbcea20afd994cf66a6d1b5b7fe77cc46bc/server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java#LL634C23-L634C23) seems not to be called...

Reply:
Looking into the changed  code here: `!Objects.equals(context.doc().getField(fieldType().name()), field)` , it would return false, because the field should be the same. 

Then the SortedSetDocValuesField would not be added.  The indexable length will be 1 instead of 2.

Reply:
[marevol](https://github.com/marevol) does changing the indexable length to 1 help pass the test? can we get this bug fix before 2.9 code freeze on July 11?

Reply:
Yes. Changing to 1 passes the test case. 
If [this fix](https://github.com/marevol/OpenSearch/commit/415b4bbcea20afd994cf66a6d1b5b7fe77cc46bc) looks okay, I'll update this PR by it.

Reply:
LGTM. Please update this PR. 

I will address the docValue in a different issue to open up a new discussion.

Reply:
Thank you for checking it. Updated.

Reply:
Other test cases failed... 
It might be better to revert dea6dbf9d537e1df80984536dd590372d3d252fb, merge this PR, and then fix the hasDocvalue problem as another problem.

Reply:
I saw the hasDocValue tests failed, so  I tried on my fork, by simply removing the getField line [here](https://github.com/mingshl/OpenSearch-Mingshl/blob/2f9728e8398e923cbbb93e36dd52c22e02612a8d/server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java#L631), it adds docValues as it was, and the length of the indexable field is still 2 and the rest of the tests are working fine. 

before pushing the PR, try running `./gradlew precommit` and `./gradlew test`, in this way you can make sure all the tests passed. 

If we are proposing to set the FlatObject hasDocValue to false, we will raise a new PR. Let's make the bug fix first then thinking about changing the docvalue.

Reply:
@marevol do you have luck in trying the above approach?

Reply:
Your change was merged into this branch, and Gradle checks have passed.